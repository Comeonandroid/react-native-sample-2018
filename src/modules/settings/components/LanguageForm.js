import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import { t } from '../../../locales/i18n'

import { View } from '../../../elements'
import { SubmitContainer } from '../../../elements'
import { BaseForm, FormContainer } from '../../../elements'
import { RadioGroup, Radio } from '../../../elements'
import { SubmitButton } from '../../../elements'

export default class LanguageForm extends PureComponent {

  constructor(props) {
    super(props)

    this.handleSubmit = this.handleSubmit.bind(this)
    this.renderForm = this.renderForm.bind(this)
  }

  handleSubmit(values, actions) {
    this.props.onSend(values)
    actions.setSubmitting(false)
  }

  renderForm(props) {
    const {
      values,
      setFieldValue,
      isSubmitting,
      handleSubmit
    } = props
    return (
      <FormContainer>
        <RadioGroup>
          {(group) => (
            <View>
              <Radio group={group} name='locale' value='cn' onChecked={setFieldValue} checked={values.locale === 'cn'}>中文</Radio>
              <Radio group={group} name='locale' value='en' onChecked={setFieldValue} checked={values.locale === 'en'}>English</Radio>
              <Radio group={group} name='locale' value='ru' onChecked={setFieldValue} checked={values.locale === 'ru'}>Русский</Radio>
            </View>
          )}
        </RadioGroup>
        <SmallSubmitContainer>
          <SubmitButton enabled={!isSubmitting} rounded onPress={handleSubmit}>{t('inputs.confirm')}</SubmitButton>
        </SmallSubmitContainer>
      </FormContainer>
    )
  }

  render() {
    return (
      <BaseForm
        initialValues={this.props.values}
        validationSchema={this.validationSchema}
        onSubmit={this.handleSubmit}
        render={this.renderForm}
      />
    )
  }
}

LanguageForm.propTypes = {
  onSend: PropTypes.func.isRequired,
  values: PropTypes.shape({
    locale: PropTypes.string
  })
}

LanguageForm.defaultProps = {
  values: {
    locale: 'en'
  }
}

const SmallSubmitContainer = styled(SubmitContainer) `
  padding: 0px 10px;
`
