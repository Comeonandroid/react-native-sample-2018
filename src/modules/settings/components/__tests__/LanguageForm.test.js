import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'

import LanguageForm from '../LanguageForm'

const setup = propOverrides => {
  const props = Object.assign({
    onSend: jest.fn(),
    values: {
      locale: ''
    }
  }, propOverrides)

  const component = <LanguageForm {...props} />
  const wrapper = shallow(component)

  return {
    props,
    wrapper,
    component
  }
}

describe('<LanguageForm />', () => {
  it('renders correctly', () => {
    const { wrapper } = setup()
    expect(wrapper).toMatchSnapshot()
  })
})
