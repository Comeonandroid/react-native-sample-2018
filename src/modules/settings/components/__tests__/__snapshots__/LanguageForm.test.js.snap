// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LanguageForm /> renders correctly 1`] = `
<BaseForm
  initialValues={
    Object {
      "locale": "",
    }
  }
  onSubmit={[Function]}
  render={[Function]}
/>
`;
